aws/rust-runtime/aws-credential-types/src/credentials_impl.rs

```diff
@@ -140,6 +140,21 @@ impl Credentials {
         )
     }
 
+    /// Creates a test `Credentials`.
+    ///
+    /// Marked as `#[doc(hidden)]` due to the usage limited to testing.
+    /// TODO(https://github.com/rust-lang/cargo/issues/2911): Consider using an auto-activated feature for testing
+    #[doc(hidden)]
+    pub fn for_tests() -> Self {
+        Self::new(
+            "ANOTREAL",
+            "notrealrnrELgWzOk3IfjzDKtFBhDby",
+            Some("notarealsessiontoken".to_string()),
+            None,
+            "test",
+        )
+    }
```

Comment:
I don't think this should be doc-hidden but we should feature gate it behind `test-util`. You can see how we handle this for `aws-smithy-client` here: https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/s3/Cargo.toml#L100-L103

Reply:
Considered once the introduction of a `test` feature but did not go for it. Happy to make the change.

Reply:
Addressed in dfecd0d.